File: Generation/CSharp/Runtime/VersionInfo.cs
// Copyright (c) Dolittle. All rights reserved.
// Licensed under the MIT license. See LICENSE file in the project root for full license information.

using Dolittle.Versioning.Contracts;

namespace Dolittle.Runtime.Contracts
{
    /// <summary>
    /// Provides information about the current version of the Runtime Contracts.
    /// </summary>
    public static class VersionInfo
    {
        /// <summary>
        /// Gets the the current <see cref="Version"/> of the Runtime Contracts.
        /// </summary>
        public static Version CurrentVersion { get; } = new Version
        {
            Major = 100,
            Minor = 200,
            Patch = 300,
            PreReleaseString = "LOCAL",
        };
    }
}

File: Generation/JavaScript/Runtime/VersionInfo.js
// Copyright (c) Dolittle. All rights reserved.
// Licensed under the MIT license. See LICENSE file in the project root for full license information.

var Fundamentals_Versioning_Version_pb = require('@dolittle/contracts/Versioning/Version_pb');

function VersionInfo() {}

VersionInfo.prototype.getCurrentVersion = function() {
    return VersionInfo.getCurrentVersion();
}

VersionInfo.getCurrentVersion = function() {
    return new Fundamentals_Versioning_Version_pb.Version()
        .setMajor(377)
        .setMinor(389)
        .setPatch(368)
        .setPrereleasestring('PRERELEASE');
}

exports.VersionInfo = VersionInfo;
